ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Util/LeakyBucket.hs

```diff
@@ -104,7 +113,7 @@ data Handlers m = Handlers
 -- first case, return the value returned by the action. In the second case,
 -- return @Nothing@.
 execAgainstBucket ::
-  (MonadDelay m, MonadAsync m, MonadFork m, MonadMask m) =>
+  (MonadDelay m, MonadAsync m, MonadFork m, MonadMask m, NoThunks (m ())) =>
```

Comment:
I think this is fine (also in the other signatures), but `IOLike m` would also work (in particular, it automatically it gives the `NoThunks` instance).

An intermediary approach would be to define
```haskell
type LeakyBucketConstraints m = (MonadDelay m, MonadAsync m, MonadFork m, MonadMask m, NoThunks (m ()))
```
and reuse that.

Reply:
Does that mean you would want me to do those changes? I quite like listing the fine-grained constraints as it gives an idea of what kind of things happen in the function very quickly, but I don't mind switching to `IOLike m` or using a type equality as suggested.

Reply:
No, just an option, the status quo also seems fine :+1: